src/addrman.cpp

```diff
@@ -562,21 +565,8 @@ void CAddrMan::Good_(const CService& addr, bool test_before_evict, int64_t nTime
     if (info.fInTried)
         return;
 
-    // find a bucket it is in now
-    int nRnd = insecure_rand.randrange(ADDRMAN_NEW_BUCKET_COUNT);
-    int nUBucket = -1;
-    for (unsigned int n = 0; n < ADDRMAN_NEW_BUCKET_COUNT; n++) {
-        int nB = (n + nRnd) % ADDRMAN_NEW_BUCKET_COUNT;
-        int nBpos = info.GetBucketPosition(nKey, true, nB);
-        if (vvNew[nB][nBpos] == nId) {
-            nUBucket = nB;
-            break;
-        }
-    }
-
-    // if no bucket is found, something bad happened;
-    // TODO: maybe re-add the node, but for now, just bail out
-    if (nUBucket == -1)
+    // if it is not in new, something bad happened
+    if (info.nRefCount == 0)
```

Comment:
I presume this shouldn't happen after the sanity checks passed on a deserialized addrman. So, on "bad things" that are caused by internal logic flaws, it might be better to crash in debug/test builds. Otherwise this will be harder to notice or only noticed later on.


```suggestion
    if (!Assume(info.nRefCount > 0)) {
```

Reply:
done as suggested